cqrs IntegrationTest

https://travis-ci.org/akka/akka-samples/jobs/618092531?utm_medium=notification&utm_source=github_status

```
java.lang.AssertionError: Expected class sample.cqrs.ShoppingCart$ItemAdded, found class sample.cqrs.ShoppingCart$ItemQuantityAdjusted (ItemQuantityAdjusted(cart-2,bar,18))
	at akka.actor.testkit.typed.internal.TestProbeImpl.assertFail(TestProbeImpl.scala:386)
	at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessageClass_internal(TestProbeImpl.scala:232)
	at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessageClass(TestProbeImpl.scala:218)
	at sample.cqrs.IntegrationTest.step03_shouldContinueEventProcessingFromOffset(IntegrationTest.java:186)
```

--- a/akka-sample-cqrs-java/src/test/java/sample/cqrs/IntegrationTest.java
+++ b/akka-sample-cqrs-java/src/test/java/sample/cqrs/IntegrationTest.java
@@ -160,8 +160,13 @@ public class IntegrationTest {
   }
 
   @Test
-  public void step03_shouldContinueEventProcessingFromOffset() {
+  public void step03_shouldContinueEventProcessingFromOffset() throws Exception {
+    // give it time to write the offset before shutting down
+    Thread.sleep(1000);
     ActorTestKit.shutdown(testKit3.system());
+
+    TestProbe<ShoppingCart.Event> eventProbe4 = testKit4.createTestProbe();
+    testKit4.system().eventStream().tell(new EventStream.Subscribe<>(ShoppingCart.Event.class, eventProbe4.getRef()));
 
     testKit4.spawn(Guardian.create(), "guardian");
 
@@ -176,9 +181,6 @@ public class IntegrationTest {
     EntityRef<ShoppingCart.Command> cart3 = ClusterSharding.get(testKit1.system())
       .entityRefFor(ShoppingCart.ENTITY_TYPE_KEY, "cart-3");
     TestProbe<ShoppingCart.Confirmation> probe3 = testKit1.createTestProbe();
-
-    TestProbe<ShoppingCart.Event> eventProbe4 = testKit4.createTestProbe();
-    testKit4.system().eventStream().tell(new EventStream.Subscribe<>(ShoppingCart.Event.class, eventProbe4.getRef()));
 
     // update from node1, consume event from node4
     cart3.tell(new ShoppingCart.AddItem("abc", 43, probe3.getRef()));

--- a/akka-sample-cqrs-scala/src/test/scala/sample/cqrs/IntegrationSpec.scala
+++ b/akka-sample-cqrs-scala/src/test/scala/sample/cqrs/IntegrationSpec.scala
@@ -143,7 +143,12 @@ class IntegrationSpec
     }
 
     "continue even processing from offset" in {
+      // give it time to write the offset before shutting down
+      Thread.sleep(1000)
       testKit3.shutdownTestKit()
+
+      val eventProbe4 = testKit4.createTestProbe[ShoppingCart.Event]()
+      testKit4.system.eventStream ! EventStream.Subscribe(eventProbe4.ref)
 
       testKit4.spawn[Nothing](Guardian(), "guardian")
 
@@ -156,9 +161,6 @@ class IntegrationSpec
 
       val cart3 = ClusterSharding(testKit1.system).entityRefFor(ShoppingCart.EntityKey, "cart-3")
       val probe3 = testKit1.createTestProbe[ShoppingCart.Confirmation]
-
-      val eventProbe4 = testKit4.createTestProbe[ShoppingCart.Event]()
-      testKit4.system.eventStream ! EventStream.Subscribe(eventProbe4.ref)
 
       // update from node1, consume event from node4
       cart3 ! ShoppingCart.AddItem("abc", 43, probe3.ref)

